source/MaterialXRuntime/Private/PvtMessage.h

```diff
@@ -18,6 +18,9 @@ enum class PvtMessageType
     RENAME_PRIM,
     REPARENT_PRIM,
     SET_ATTRIBUTE,
+    SET_METADATA,
+    CREATE_METADATA,
+    REMOVE_METADATA,
```

Comment:
I don't think you need to CREATE message. The SET message should cover that. And the set message is what is being sent above when creating a new metadata (it seems the only place where create message is sent is when remove-undo is performed, which seems inconsistent). So better to use only SET and REMOVE consistently.